src/hotspot/share/opto/library_call.cpp

```diff
@@ -3767,10 +3765,20 @@ Node* LibraryCallKit::generate_access_flags_guard(Node* kls, int modifier_mask,
   return generate_fair_guard(bol, region);
 }
 Node* LibraryCallKit::generate_interface_guard(Node* kls, RegionNode* region) {
-  return generate_access_flags_guard(kls, JVM_ACC_INTERFACE, 0, region);
+  Node* modp = basic_plus_adr(kls, in_bytes(Klass::access_flags_offset()));
+  Node* mods = make_load(nullptr, modp, TypeInt::INT, T_INT, MemNode::unordered);
+  return generate_mods_flags_guard(mods, JVM_ACC_INTERFACE, 0, region);
 }
+
+// Use this for testing if Klass is_hidden, has_finalizer, and is_cloneable_fast.
+Node* LibraryCallKit::generate_misc_flags_guard(Node* kls, int modifier_mask, int modifier_bits, RegionNode* region) {
+  Node* p = basic_plus_adr(kls, in_bytes(Klass::misc_flags_offset()));
+  Node* mods = make_load(nullptr, p, TypeInt::UBYTE, T_BOOLEAN, MemNode::unordered);
+  return generate_mods_flags_guard(mods, modifier_mask, modifier_bits, region);
```

Comment:
```suggestion
  return generate_mods_flags_guard(mods, modifier_mask, modifier_bits, region, Klass::misc_flags_offset(), TypeInt::UBYTE, T_BOOLEAN);
```
This looks much better, but can't you leave the basic_plus_adr and make_load in generate_mods_flags_guard, and pass in the needed specialization?

Reply:
Really, this is better? it adds three parameters.

Reply:
It reduces duplicate code, which is usually good.  Yes, I like it better.

Reply:
Ok!